inc/Qdrant_API.php

```diff
@@ -0,0 +1,422 @@
+				( new Filter() )->addMust(
+					new MatchString( 'post_id', (string) $id )
+				)
+			);
+
+			return 'acknowledged' === $response['result']['status'];
+		} catch ( \Exception $e ) {
+			if ( 403 === $e->getCode() ) {
+				update_option( 'hyve_qdrant_status', 'inactive' );
+			}
+
+			return new \WP_Error( 'collection_error', $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Search collection.
+	 * 
+	 * @param array $embeddings Embeddings.
+	 * 
+	 * @return array|\WP_Error
+	 */
+	public function search( $embeddings ) {
+		try {
+			$search = (
+				new SearchRequest(
+					new VectorStruct( $embeddings, 'embeddings' )
+				)
+			)
+			->setLimit( 10 )
+			->setWithPayload( true );
+
+			$response = $this->client->collections( self::COLLECTION_NAME )->points()->search( $search );
+
+			if ( empty( $response['result'] ) ) {
+				return [];
+			}
+
+			$results = $response['result'];
+
+			$payload = array_map(
+				function ( $result ) {
+					$payload          = $result['payload'];
+					$payload['score'] = $result['score'];
+					return $payload;
+				},
+				$results
+			);
+
+			return $payload;
+		} catch ( \Exception $e ) {
+			if ( 403 === $e->getCode() ) {
+				update_option( 'hyve_qdrant_status', 'inactive' );
+			}
+
+			return new \WP_Error( 'collection_error', $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Disconnect Qdrant Integration.
+	 *
+	 * @since 1.3.0
+	 *  
+	 * @return bool|\WP_Error
+	 */
+	public function disconnect() {
+		try {
+			$response = $this->client->collections( self::COLLECTION_NAME )->points()->deleteByFilter(
+				( new Filter() )->addMust(
+					new MatchString( 'website_url', get_site_url() )
+				)
+			);
+
+			return 'acknowledged' === $response['result']['status'];
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'collection_error', $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Migrate Data to Qdrant.
+	 * 
+	 * @return void
+	 */
+	public function migrate_data() {
+		$db_table = DB_Table::instance();
+		$posts    = $db_table->get_by_storage( 'WordPress' );
+
+		if ( empty( $posts ) ) {
+			return;
+		}
+
+		$points = [];
+
+		foreach ( $posts as $post ) {
+			$points[] = [
+				'embeddings' => json_decode( $post->embeddings, true ),
+				'data'       => [
+					'post_id'      => $post->post_id,
+					'post_title'   => $post->post_title,
+					'post_content' => $post->post_content,
+					'token_count'  => $post->token_count,
+					'website_url'  => get_site_url(),
+				],
+			];
+		}
+
+		$qdrant = self::instance();
+
+		$success = $qdrant->add_points( $points );
+
+		if ( is_wp_error( $success ) ) {
+			return;
+		}
+
+		foreach ( $posts as $post ) {
+			$db_table->update(
+				$post->id,
+				[
+					'storage' => 'Qdrant',
+				] 
+			);
+		}
+
+		$migration_status = get_option( 'hyve_qdrant_migration', [] );
+
+		$migration_status['current'] += count( $posts );
+
+		$current  = $migration_status['current'] ?? 0;
+		$total    = $migration_status['total'] ?? 0;
```

Comment:
Since the progress bar computes the progress as `current/max`, having max with possibility to be `0` is problematic. In the component `ProgressBar`, you used `?.max || 100` to counter this problem.

But this is a self-inflicted problem. The backend should provide the proper values. Since we control both sides, the front end should not fix the backend's problem. 

I suggest `$total = $migration_status['total'] ?? 100;`

Reply:
Here the total is not the percent but the number of posts, so we don't want to put 100 there if there are no posts to migrate That condition there is more of a safeguard in case there is some issue with the option to avoid errors.